packages/markers/src/browser/problem/problem-widget.tsx

```diff
@@ -81,6 +83,43 @@ export class ProblemWidget extends TreeWidget {
         return 'caption';
     }
 
+    protected renderTailDecorations(node: TreeNode, props: NodeProps) {
+        return <div className='row-button-container'>
+            {this.renderRemoveButton(node)}
+        </div>;
+    }
+
+    protected renderRemoveButton(node: TreeNode): React.ReactNode {
+        return <span className='remove-node' onClick={e => this.remove(node, e)}></span>;
+    }
+
+    protected readonly remove = (node: TreeNode, e: React.MouseEvent<HTMLElement>) => this.doRemove(node, e);
+    protected doRemove(node: TreeNode, e: React.MouseEvent<HTMLElement>) {
+        this.removeNode(node);
+        e.stopPropagation();
+    }
+
+    protected removeNode(node: TreeNode) {
```

Comment:
i think this method should belong to `ProblemTreeModel` and look like that without changing any other apis:
```ts
    protected removeNode(node: TreeNode) {
        if (MarkerInfoNode.is(node)) {
            this.problemManager.cleanAllMarkers(node.uri);
        }
        if (MarkerNode.is(node)) {
            const { uri } = node;
            const { owner } = node.marker;
            const diagnostics = this.problemManager.findMarkers({ uri, owner, dataFilter: data => node.marker.data !== data }).map(({ data }) => data);
            this.problemManager.setMarkers(uri, owner, diagnostics);
        }
    }
```